### PR TITLE
Backport XCM fixes to 1.5.0

### PR DESCRIPTION
**Pull request by franciscoaguirre**
- Transactional processing
- Delivery fees taken from holding

### CODE REVIEW

File: polkadot/xcm/xcm-executor/src/lib.rs [hunk 1]
```diff
@@ -630,9 +630,39 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				Ok(())
 			},
 			DepositReserveAsset { assets, dest, xcm } => {
-				let deposited = self.holding.saturating_take(assets);
-				for asset in deposited.assets_iter() {
-					Config::AssetTransactor::deposit_asset(&asset, &dest, Some(&self.context))?;
+				let old_holding = self.holding.clone();
+				let result = Config::TransactionalProcessor::process(|| {
```
Comment (acatangiu) [inline polkadot/xcm/xcm-executor/src/lib.rs:634]: This adds both transactionality and delivery-fees-fix to `DepositReserveAsset`, shouldn't we also backport transactionality for all others too?

Comment (franciscoaguirre) [reply polkadot/xcm/xcm-executor/src/lib.rs:708]: Yes, cherry picked the other commit

**Comment by franciscoaguirre** [comment]
bot fmt

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5100317

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5100352

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5100333

**Comment by xlc** [comment]
will there be one to back port to 1.6.0 or it is already done?

File: cumulus/parachains/runtimes/coretime/coretime-westend/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,853 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
```
Comment (acatangiu) [inline cumulus/parachains/runtimes/coretime/coretime-westend/src/lib.rs:1]: you accidentally brought in the new coretime chain, please remove from this PR

File: cumulus/parachains/runtimes/people/people-rococo/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,838 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
```
Comment (acatangiu) [inline cumulus/parachains/runtimes/people/people-rococo/src/lib.rs:1]: you accidentally brought in the new people chain, please remove from this PR

File: cumulus/parachains/runtimes/people/people-westend/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,838 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
```
Comment (acatangiu) [inline cumulus/parachains/runtimes/people/people-westend/src/lib.rs:1]: you accidentally brought in the new people chain, please remove from this PR

**Review by acatangiu — CHANGES_REQUESTED** [review]

**Comment by franciscoaguirre** [comment]
> will there be one to back port to 1.6.0 or it is already done?

Yes, there will be

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5105684

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5105720

**Comment by bkontur** [comment]
@franciscoaguirre you need at least these two commits (xcm benchmarks fix + rust 1.7.4) - see last two commits in my backport to 1.5.0 https://github.com/paritytech/polkadot-sdk/pull/3094/commits

File: polkadot/xcm/xcm-executor/src/lib.rs [hunk 5]
```diff
@@ -456,18 +507,6 @@ impl<Config: config::Config> XcmExecutor<Config> {
 		Ok(())
 	}
 
-	/// Refund any unused weight.
-	fn refund_surplus(&mut self) -> Result<(), XcmError> {
```
Comment (acatangiu) [inline polkadot/xcm/xcm-executor/src/lib.rs:460]: nit:

I think `fn subsume_asset()` and `fn subsume_assets()` can also be removed

**Review by acatangiu — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5107247

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5107282

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5107265

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5109743

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5110449

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5117382

**Comment by acatangiu** [comment]
hmm.. what are the CI issues currently blocking us here? is it the upgraded toolchain in CI vs old codebase?

**Comment by bkontur** [comment]
@EgorPopelyaev 
is target branch ok?  `release-crates-io-v1.5.0` vs `release-polkadot-v1.5.0`?
this  https://github.com/paritytech/polkadot-sdk/pull/3094 was merged to the `release-polkadot-v1.5.0`

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-slow
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5152431

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5152424

**Pull request merged**